packages/wrangler/src/module-collection.ts

```diff
@@ -29,55 +93,78 @@ export default function makeModuleCollector(props: {
-              watchFiles: [filePath], // we also add the file to esbuild's watch list
-            };
-          }
-        );
+                // add the module to the array
+                modules.push({
+                  name: fileName,
+                  content: fileContent,
+                  type: RuleTypeToModuleType[rule.type],
+                });
 
-        if (props.format !== "modules") {
-          build.onLoad(
-            { filter: /.*\.(wasm)$/ },
-            async (args: esbuild.OnLoadArgs) => {
-              return {
-                // We replace the the wasm module with an identifier
-                // that we'll separately add to the form upload
-                // as part of [wasm_modules]. This identifier has to be a valid
-                // JS identifier, so we replace all non alphanumeric characters
-                // with an underscore.
-                contents: `export default ${args.path.replace(
-                  /[^a-zA-Z0-9_$]/g,
-                  "_"
-                )};`,
-              };
+                return {
+                  path: fileName, // change the reference to the changed module
+                  external: props.format === "modules", // mark it as external in the bundle
+                  namespace: `wrangler-module-${rule.type}`, // just a tag, this isn't strictly necessary
+                  watchFiles: [filePath], // we also add the file to esbuild's watch list
+                };
+              }
+            );
+
+            if (props.format === "service-worker") {
+              build.onLoad(
+                { filter: globToRegExp(glob) },
+                async (args: esbuild.OnLoadArgs) => {
+                  return {
+                    // We replace the the module with an identifier
+                    // that we'll separately add to the form upload
+                    // as part of [wasm_modules]/[text_blobs]. This identifier has to be a valid
+                    // JS identifier, so we replace all non alphanumeric characters
+                    // with an underscore.
+                    // TODO: what of "Data"?
+                    contents: `export default ${args.path.replace(
+                      /[^a-zA-Z0-9_$]/g,
+                      "_"
+                    )};`,
+                  };
+                }
+              );
             }
-          );
-        }
+          });
+        });
+
+        rulesToRemove.forEach((rule) => {
+          rule.globs.forEach((glob) => {
+            build.onResolve(
+              { filter: globToRegExp(glob) },
+              async (args: esbuild.OnResolveArgs) => {
+                throw new Error(
+                  `The file ${
+                    args.path
+                  } matched a module rule in your configuration (${JSON.stringify(
+                    rule
+                  )}), but was ignored because a previous rule with the same type had \`fallthrough = false\`.`
```

Comment:
Strictly, I think it is ignored because a previous rule did not have `fallthrough = true`, right?

Reply:
Good catch, yes.